presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java

```diff
@@ -785,7 +786,18 @@ else if (preparedQuery.isExplainTypeValidate()) {
                 log.error(eventFailure, "Error publishing query immediate failure event");
             }
 
-            throw toPrestoSparkFailure(session, failureInfo.get());
+            if (isFatalException(executionFailure)) {
```

Comment:
Maybe use a separate catch(PrestoSparkFatalException e) for readability.

Reply:
Just checked, we still need previous catch (on Throwable (will be applied for both fatal and non-fatal error) to do post-processing/clean up) and then re-throw the exception here, so have to keep this.